include/named_threads.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <string>
+#include <thread>
+
+namespace celerity::detail {
+
+std::thread::native_handle_type get_current_thread_handle();
+
+void set_thread_name(const std::thread::native_handle_type thread_handle, const std::string& name);
```

Comment:
Pedantic, just because clang-tidy complains about this: top-level cv-qualifiers on parameters are not part of the function signature and can be removed in prototypes since they are an implementation detail.